### PR TITLE
Various functions in the 16C8E0+ region

### PR DESCRIPTION
**Pull request by ethteck**

### CODE REVIEW

File: src/code_1A5830.c [hunk 1]
```diff
@@ -1,12 +1,51 @@
...
+
+    while (partIt != NULL) {
+        if (partIt->eventFlags & 0xC0000) {
+            ret = TRUE;
+            break;
+        } else {
+            partIt = partIt->nextPart;
+        }
+    }
+    return ret;
```
Comment (nanaian) [inline src/code_1A5830.c:17]: newline before return

File: src/code_1A5830.c [hunk 2]
```diff
@@ -1,12 +1,51 @@
...
+void dispatch_event_actor(Actor* actor, Event event);
+
+s32 func_80276F50(Actor* actor) {
+    ActorPart* partIt = actor->partsTable;
+    s32 ret = FALSE;
+
+    while (partIt != NULL) {
+        if (partIt->eventFlags & 0xC0000) {
+            ret = TRUE;
+            break;
```
Comment (nanaian) [inline src/code_1A5830.c:12]: `return TRUE` doesn't match? why did they write it like this lol

File: src/code_1A5830.c [hunk 3]
```diff
@@ -392,7 +465,19 @@ ApiStatus GetLastDamage(ScriptInstance* script, s32 isInitialCall) {
 
 INCLUDE_ASM(s32, "code_1A5830", EnableActorGlow);
 
+// Cannot get gBattleStatus to load via lui+addiu no matter what I do
```
Comment (nanaian) [inline src/code_1A5830.c:468]: Probably a .data issue? i.e., gBattleStatus is a global variable defined in this file [it'd make sense, this is the battle engine file]

File: src/code_197F40.c [hunk 4]
```diff
@@ -9,17 +9,56 @@ s8 count_targets(Actor* actor, s32 targetHomeIndex, s32 targetSelectionFlags) {
     return actor->targetListLength;
 }
 
-INCLUDE_ASM(s32, "code_197F40", get_nearest_home_index);
+s32 get_nearest_home_index(f32 x, f32 y, f32 z) {
+    s32 xVal;
+    s32 yVal;
+
+    xVal = 0;
+    if (!(y < 40.0f)) {
```
[the remark this answers is not in the document]
Comment (ethteck) [reply src/code_197F40.c:17]: It doesn't match. There may be a better way to re-write this whole function, but I didn't want to mess with it..

File: src/code_1A5830.c (hunk 3 above)
Comment (ethteck) [reply src/code_1A5830.c:468]: oh! hmm yeah that is possible then!

File: src/code_197F40.c (hunk 4 above)
[the remark this answers is not in the document]
Comment (nanaian) [reply src/code_197F40.c:17]: @pixel-stuck probably knows how

File: src/code_16c8e0.c [hunk 5]
```diff
@@ -1,6 +1,42 @@
 #include "common.h"
 
-INCLUDE_ASM(s32, "code_16c8e0", get_dpad_input_radial);
+void get_dpad_input_radial(f32* angle, f32* magnitude) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    u16 currentButtonsDown;
+    f32 stickX;
+    f32 stickY;
+    f32 mag;
+    f32 sixty = 60.0f;
```
[the remark this answers is not in the document]
Comment (ethteck) [reply src/code_16c8e0.c:9]: It is required unfortunately, but yeah I can rename it

File: include/common_structs.h [hunk 6]
```diff
@@ -1202,9 +1202,7 @@ typedef struct Actor {
     /* 0x000 */ s32 flags;
     /* 0x004 */ char unk_04[4];
     /* 0x008 */ struct StaticActorData* staticActorData;
-    /* 0x00C */ Vec3f moveCurrentPos;
-    /* 0x018 */ Vec3f moveGoalPos;
-    /* 0x024 */ Vec3f moveEndPos; /* where other actors should target this one at. saved before partner switching */
+    /* 0x00C */ Vec3f movePos[3]; // current, goal, end
```
[the remark this answers is not in the document]
Comment (ethteck) [reply include/common_structs.h:1205]: Yeah, possibly. I'll try it

File: src/code_16c8e0.c [hunk 7]
```diff
@@ -35,4 +78,40 @@ INCLUDE_ASM(s32, "code_16c8e0", func_80240AA8);
 
 INCLUDE_ASM(s32, "code_16c8e0", delete_actor);
 
-INCLUDE_ASM(s32, "code_16c8e0", delete_player_actor);
+void delete_player_actor(Actor* player) {
+    s32 i;
```
[the remark this answers is not in the document]
Comment (ethteck) [reply src/code_16c8e0.c:82]: I usually declare things in order of use, which is why I put it there

[the remark this answers is not in the document]
Comment (nanaian) [reply src/code_16c8e0.c:82]: That makes sense. In a case like this function I'd opt for using blocks and declaring temporaries within the blocks as they're required, but it doesn't really matter all that much.

File: include/functions.h [hunk 8]
```diff
@@ -95,6 +95,7 @@ void func_800ED5D0(Npc* partner);
 
 s32 func_800EF394(f32);
 s32 does_script_exist(s32 id);
+s32 does_script_exist_by_ref(ScriptInstance* script);
```
[the remark this answers is not in the document]
Comment (ethteck) [reply include/functions.h:98]: Well they exist in a different c file from si.c, so I'm not sure they fall under the scope of "script interpreter" but are more generally just pertaining to scripts

File: src/code_1A5830.c (hunk 2 above)
Comment (ethteck) [reply src/code_1A5830.c:12]: It doesn't...yeah lol

File: include/functions.h (hunk 8 above)
[the remark this answers is not in the document]
Comment (nanaian) [reply include/functions.h:98]: Oh I assumed they lived in si.c, nvm

**Review by pixel-stuck — APPROVED** [review]
didn't see anything, lgtm

**Pull request merged**